### DISCUSSION TITLE
How to force a library to be part of the group?

### DISCUSSION THREAD

**Discussion by nZeus** [renovatebot/renovate#25979] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

39.1.3

### If you're self-hosting Renovate, select which platform you are using.

github.com

### What is your question?

Hello,

We have the following config file with 2 groups - angular-non-major and angular-major:

```
    {
      "matchManagers": [
        "npm"
      ],
      "groupName": "angular monorepo (non-major)",
      "matchUpdateTypes": [
        "minor",
        "patch"
      ],
      "matchDepTypes": [
        "dependencies",
        "devDependencies"
      ],
      "matchPackagePatterns": [
        "angular",
        "@ngrx",
        "zone.js"
      ],
      "automerge": true,
      "automergeType": "pr"
    },

    {
      "matchManagers": [
        "npm"
      ],
      "groupName": "angular monorepo (major)",
      "matchUpdateTypes": [
        "major"
      ],
      "matchDepTypes": [
        "dependencies",
        "devDependencies"
      ],
      "matchPackagePatterns": [
        "angular",
        "@ngrx",
        "zone.js"
      ]
    },
```

We configured angular monorepo group to include `angular`, `@ngrx` and `zone.js` packages.
Renovate bot has detected a *major* upgrade for angular libraries (`16 -> 17`). It also detected a new version of `zone.js` (`0.13.3 -> 0.14`), but that one has a minor upgrade, to it didn't add it to angular-major PR. 
Instead, it creates 2 PRs:

<img width="872" alt="image" src="https://github.com/renovatebot/renovate/assets/2727326/f1fa61ab-8908-4f19-a0d2-de822f9b2bbf">

Both PRs are failing, because libraries have to be upgraded together.
Is there a way to configure renovate-bot to _always_ keep these libraries together?

Kind regards,
Ilia

**Comment by nZeus** [comment]
Just tried it with a single group

```
    {
      "matchManagers": [
        "npm"
      ],
      "groupName": "angular monorepo",
      "matchUpdateTypes": [
        "major",
        "minor",
        "patch"
      ],
      "matchDepTypes": [
        "dependencies",
        "devDependencies"
      ],
      "matchPackagePatterns": [
        "angular",
        "@ngrx",
        "zone.js"
      ]
    },
```
, but still it generates 2 PRs, which is incorrect... 

|PR1|PR2|
|-|-|
|<img width="864" alt="image" src="https://github.com/renovatebot/renovate/assets/2727326/0f504318-dc4f-4195-a2ca-47a34b725968">|<img width="741" alt="image" src="https://github.com/renovatebot/renovate/assets/2727326/36654b06-3ed7-4ddc-bea7-4c0b8a506bc5">|

**Comment by rarkins** [comment]
Renovate splits major and non-major groups by default. If you look at the branch names, you can see the left side screenshot is major updates and the right-side is non-major. You could manualy coerce the right side dependency into the same group by specifying the `groupSlug` for it as `major-angular-monorepo`.

**Reply by nZeus** [reply]
I tried that:
<img width="353" alt="image" src="https://github.com/renovatebot/renovate/assets/2727326/319ea2ca-5b44-4ae3-9ef4-d41f53fd106b">

but renovate also adds the `major-` prefix to the branch, so it still creates 2 PRs:
<img width="548" alt="image" src="https://github.com/renovatebot/renovate/assets/2727326/b9bb8e62-2151-4bc4-9d5c-a1b9cee47d9a">

<img width="1168" alt="image" src="https://github.com/renovatebot/renovate/assets/2727326/5d7f1a02-fe85-4583-bee1-83fa4fe869de">

**Reply by rarkins** [reply, marked as answer]
Set the groupSlug only for that one specific dependency on the right

**Reply by nZeus** [reply]
Indeed, this trick worked well!

Thank you!
